doc/sphinx/addendum/generalized-rewriting.rst

```diff
@@ -834,17 +868,6 @@ are applied using the tactic :n:`rewrite_strat @rewstrategy`.
    to be documented
 
 
-A few of these are defined in terms of the others using a
```

Comment:
I think it's important to not drop this line (and the one after it), since the primitive fixpoint operator isn't actually exposed.

Reply:
Updated.  I reverted moving these descriptions.

The original text leaves much to be desired:
- It's not obvious that `fix` corresponds to "primitive fixpoint operator"
- No description of `fix`
- It's perverse to give equivalences using an operator that's not available to users.
- The description for `@one_term` only mentions lemmas, not fixpoint values
- `fix @ident.` probably should have a semicolon rather than a period (?)

If it's in the code, it appears to work and it's useful, would you care to file an issue asking to make it accessible and documented?

I don't see how `any @rewstrategy := fix @ident. try (@rewstrategy ; @ident)` would cause repetition.

Should `try @rewstrategy := choice @rewstrategy id` be `try @rewstrategy := fix @ident; choice @rewstrategy @ident`?  If so, I will fix that (and periods -> semicolons).

Reply:
@jfehrle 

> * It's not obvious that `fix` corresponds to "primitive fixpoint operator"

Indeed, though `fix` is Gallina's primitive fixpoint operator, so it's not much of a stretch to imagine a similar construct to Gallina `fix` for rewrite strategies

> - No description of `fix`

To run `fix <ident>. <body>`, replace all occurrences of `<ident>` in `<body>` with `fix <ident>. <body>` and then run the resulting code.

> - It's perverse to give equivalences using an operator that's not available to users.

I'd say unfortunate.  It is very informative to users like me, but it would be better if it were available.


> * The description for `@one_term` only mentions lemmas, not fixpoint values

Yes, this is equivalent to fixpoint values not being user accessible.

> * `fix @ident.` probably should have a semicolon rather than a period (?)

No, treat `fix` like a lambda.  Often in math λ arguments are followed by `.`, sometimes by `,`, sometimes by `=>` or `->` or `⇒` or `→` or `↦`, but I've never seen them followed by a `;`

> If it's in the code, it appears to work and it's useful, would you care to file an issue asking to make it accessible and documented?

https://github.com/coq/coq/issues/13702

> I don't see how `any @rewstrategy := fix @ident. try (@rewstrategy ; @ident)` would cause repetition.

Using "⇝" to mean "evaluates to", we have `any @rewstrategy` ⇝ `fix @ident. try (@rewstrategy ; @ident)` ⇝ `try (@rewstrategy ; (fix @ident. try (@rewstrategy ; @ident)))` ⇝ `try (@rewstrategy ; try (@rewstrategy ; (fix @ident. try (@rewstrategy ; @ident))))`.  There's the source of repetition.

> Should `try @rewstrategy := choice @rewstrategy id` be `try @rewstrategy := fix @ident; choice @rewstrategy @ident`?  If so, I will fix that (and periods -> semicolons).

No.  Your code gives an infinite loop on failure of `@rewstrategy`, because the fixpoint here is `choice @rewstrategy (choice @rewstrategy (choice @rewstrategy (choice @rewstrategy ...)))`, and we really want `choice @rewstrategy id` (i.e., act as the identity when `@rewstrategy` fails).

Reply:
Thanks for explaining and the review.  My goal for the long term is to make the doc understandable to readers who only have a bachelor's degree in math or CS.  But lots to do to get there...